### PR TITLE
Changed the way to store assignee, mention, reporter and watching in struct so that the code can be easily tested

### PR DESCRIPTION
**Pull request by Nityanand13**

### CODE REVIEW

File: server/issue.go [hunk 1]
```diff
@@ -1094,7 +1094,7 @@ func (p *Plugin) applyReporterNotification(wh *webhook, instanceID types.ID, rep
 	commentMessage := fmt.Sprintf("%s **commented** on %s:\n> %s", commentAuthor, jwhook.mdKeySummaryLink(), jwhook.Comment.Body)
 
 	connection, err := p.GetUserSetting(wh, instanceID, reporter.Name, reporter.AccountID)
-	if err != nil || connection.Settings == nil || !connection.Settings.ShouldReceiveNotificationsForReporter() {
+	if err != nil || connection.Settings == nil || !connection.Settings.ShouldReceiveNotificationsFor("reporter") {
```
Comment (Kshitij-Katiyar) [inline server/issue.go:1097]: create a constant for reporter

File: server/issue.go [hunk 2]
```diff
@@ -1131,39 +1131,15 @@ func (p *Plugin) GetUserSetting(wh *webhook, instanceID types.ID, jiraAccountID,
 	return connection, nil
 }
 
-func (s *ConnectionSettings) ShouldReceiveNotificationsForAssignee() bool {
-	if s.SendNotificationsForAssignee != nil {
-		return *s.SendNotificationsForAssignee
+func (s *ConnectionSettings) ShouldReceiveNotificationsFor(role string) bool {
```
Comment (Kshitij-Katiyar) [inline server/issue.go:1134]:
```suggestion
func (s *ConnectionSettings) ShouldReceiveNotification(role string) bool {
```

File: server/command_test.go [hunk 3]
```diff
@@ -72,11 +72,13 @@ func getMockUserStoreKV() mockUserStoreKV {
 	trueValue := true
 	withNotifications := connection // copy
 	withNotifications.Settings = &ConnectionSettings{
-		Notifications:                trueValue,
-		SendNotificationsForMention:  &trueValue,
-		SendNotificationsForAssignee: &trueValue,
-		SendNotificationsForReporter: &trueValue,
-		SendNotificationsForWatching: &trueValue,
+		Notifications: trueValue,
+		RoleNotification: map[string]*bool{
```
Comment (Kshitij-Katiyar) [inline server/command_test.go:76]:
```suggestion
		RolesForDMNotification: map[string]*bool{
```

File: server/settings.go [hunk 4]
```diff
@@ -20,6 +20,13 @@ const (
 	subCommandWatching  = "watching"
 )
 
+func (connection *Connection) sendNotificationsFor(role string, hasNotification bool) bool {
```
Comment (Kshitij-Katiyar) [inline server/settings.go:23]:
```suggestion
func (connection *Connection) sendNotification(role string, hasNotification bool) bool {
```

File: server/settings.go [hunk 5]
```diff
@@ -20,6 +20,13 @@ const (
...
+	if role != "assignee" && role != "mention" && role != "reporter" && role != "watching" {
```
Comment (Kshitij-Katiyar) [inline server/settings.go:24]: constants for all these

File: server/user_server.go [hunk 6]
```diff
@@ -105,11 +105,13 @@ func (p *Plugin) httpOAuth1aComplete(w http.ResponseWriter, r *http.Request, ins
 	trueValue := true
 	// Set default settings the first time a user connects
 	connection.Settings = &ConnectionSettings{
-		Notifications:                trueValue,
-		SendNotificationsForWatching: &trueValue,
-		SendNotificationsForMention:  &trueValue,
-		SendNotificationsForAssignee: &trueValue,
-		SendNotificationsForReporter: &trueValue,
+		Notifications: trueValue,
+		RoleNotification: map[string]*bool{
```
Comment (Kshitij-Katiyar) [inline server/user_server.go:109]: same as above and give detailed variable names everywhere

**Review by Kshitij-Katiyar — APPROVED** [review]

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**